idseq_dag/util/command_patterns.py

```diff
@@ -71,6 +72,7 @@ def open(self, stdin: int = None, stdout: int = None, stderr: int = None) -> sub
             self._command_args(),
             shell=False,
             cwd=self.cd,
+            env=os.environ if self.env == None else self.env,
```

Comment:
@davidrissato What do you think of passing environment to subcommands in this manner?  Any issues?

Reply:
We are pretty sure that the "env" contents would not be sent to the log, correct?

Reply:
Yes, that's correct. Only the scripts, variables and directory should be logged. This is controlled here:
https://github.com/chanzuckerberg/idseq-dag/blob/348f718c9834808008dbc15e96bdfdac8af5c8a5/idseq_dag/util/command_patterns.py#L227-L233

Reply:
I believe we may actually want to merge `os.environment` with `self.env`, like this:
```
     env = dict(os.environ, **(self.env or {}))
```

Reply:
Thanks, @davidrissato   I agree that most often in practice we want to merge (so as to add or override a few variables) rather than replace the entire environment.

However, I wanted to leave that to the caller -- just like the python library designers have chosen to do.  The merge step you suggest is implemented by the caller.

I think as a library writer I probably would have preferred to supply a boolean flag to say whether I want the environment merged or replaced, with the default being merged, but alas, the python library writers didn't choose that approach, and I fear going contrary to their decision could confuse some people.